test/lib/global_uid_test.rb

```diff
@@ -472,13 +472,53 @@ def with_timed_out_connection(server:, end_time:)
           this_id = WithGlobalUID.create!.id
           assert_operator this_id, :>, last_id
         end
+
+        # The same allocation server is used each time `with_servers` is called
+        refute_empty(GlobalUid::Base.servers[0].send(:allocator).recent_allocations)
+        assert_empty(GlobalUid::Base.servers[1].send(:allocator).recent_allocations)
+      end
+
+      it 'still selects a connection at random on initialization' do
+        server_init_order = 10.times.map { GlobalUid::Base.init_server_info.map(&:name) }
+
+        refute(server_init_order.all? { |order| order == ["test_id_server_1", "test_id_server_2"] } )
+        assert(server_init_order.any? { |order| order == ["test_id_server_1", "test_id_server_2"] } )
+        assert(server_init_order.any? { |order| order == ["test_id_server_2", "test_id_server_1"] } )
```

Comment:
Won’t either of these two assertions on `any?` fail once every 2^10 runs and the gem randomly selects the same server ten times in a row?

Reply:
Yes, there's a low chance it'll fail. I'll add a similar comment to what we've done with the others.  It's better than leaving out the `assert` statements which clearly demonstrate what's happening with the order.